api/src/main/kotlin/io/spine/protodata/event/CompilerEvent.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata.event
+
+import io.spine.base.EventMessage
+
+/**
+ * A message of a Protobuf compiler event.
+ */
+public interface CompilerEvent : EventMessage {
+
+    /**
+     * Shows if this event should result in code generation.
+     *
+     * If `false`, the event is purely informative and should not add any code generation tasks.
+     * Such events are emitted for the Proto definitions that come from a module's dependency.
+     *
+     * If `true`, the event describes a Proto definition that belongs to the currently analyzed
+     * module and thus may result in code generation.
+     */
+    public fun getGenerationRequested(): Boolean
```

Comment:
If we take all the `CompilerEvent`s, what percentage of them would be non-generative?

I'm asking to see if we may way to make it vice-versa, e.g. have `isInformative()`/`getInformative()` instead.

Reply:
This way could probably reduce the number of setting the flag for enabling code generation.

Reply:
When setting the flag, it's not relevant if the value is `true` or `false`, the number of operations will be the same. As per the percentage, I'd assume that on bigger projects `true` would win out over `false`. But it's impossible to tell in the general case.